msal/src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -87,29 +87,14 @@ internal class AuthenticationRequestParameters
                 }
                 else
                 {
-                    if (ClientCredential.Assertion == null || ClientCredential.ValidTo != 0)
+                    if (ClientCredential.Certificate != null)
                     {
-                        bool assertionNearExpiry = (ClientCredential.ValidTo <=
-                                                    Jwt.JsonWebToken.ConvertToTimeT(DateTime.UtcNow +
-                                                                                    TimeSpan.FromMinutes(
-                                                                                    Constants 
-                                                                                    .ExpirationMarginInMinutes)));
-                        if (assertionNearExpiry)
```

Comment:
@jmprieur Can you think of any reasons as to why we would want to cache a signed client assertion for 10 minutes instead of resigning it for each authentication request in MSAL? ADAL does not have this caching

Reply:
@trwalke : no. Do we understand how the code went to MSAL.NET ? I don't think that we tested the OBO flow yet (as it's only recently been available on the v2.0 endpoint)

Reply:
maybe we should ask @SomkaPe as well?

Reply:
@jmprieur I checked an old PR (https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/191) and it did not mention anything specific about this. I also talked with Jason who reviewed the PR and he did not remember anything about it as well. So at this point we are not sure. However,  I see no value in it as it will only be useful when there are a significant amount of authentication requests in a short period of time which I do not believe to be a very common use case for the client credential flow.